common/djangoapps/student/tests/test_linkedin.py

```diff
@@ -2,120 +2,101 @@
-from student.models import LinkedInAddToProfileConfiguration
 
 
 @ddt.ddt
 class LinkedInAddToProfileUrlTests(TestCase):
     """Tests for URL generation of LinkedInAddToProfileConfig. """
 
-    COURSE_KEY = CourseLocator(org="edx", course="DemoX", run="Demo_Course")
-    COURSE_NAME = u"Test Course ☃"
-    CERT_URL = u"http://s3.edx/cert"
+    COURSE_NAME = 'Test Course ☃'
+    CERT_URL = 'http://s3.edx/cert'
     SITE_CONFIGURATION = {
         'SOCIAL_SHARING_SETTINGS': {
             'CERTIFICATE_LINKEDIN_MODE_TO_CERT_NAME': {
-                'honor': u'{platform_name} Honor Code Credential for {course_name}',
-                'verified': u'{platform_name} Verified Credential for {course_name}',
-                'professional': u'{platform_name} Professional Credential for {course_name}',
-                'no-id-professional': u'{platform_name} Professional Credential for {course_name}',
+                'honor': '{platform_name} Honor Code Credential for {course_name}',
+                'verified': '{platform_name} Verified Credential for {course_name}',
+                'professional': '{platform_name} Professional Credential for {course_name}',
+                'no-id-professional': '{platform_name} Professional Credential for {course_name}',
             }
         }
     }
 
     @ddt.data(
-        ('honor', u'Honor+Code+Certificate+for+Test+Course+%E2%98%83'),
-        ('verified', u'Verified+Certificate+for+Test+Course+%E2%98%83'),
-        ('professional', u'Professional+Certificate+for+Test+Course+%E2%98%83'),
-        ('default_mode', u'Certificate+for+Test+Course+%E2%98%83')
+        ('honor', 'Honor+Code+Certificate+for+Test+Course+%E2%98%83'),
+        ('verified', 'Verified+Certificate+for+Test+Course+%E2%98%83'),
+        ('professional', 'Professional+Certificate+for+Test+Course+%E2%98%83'),
+        ('default_mode', 'Certificate+for+Test+Course+%E2%98%83')
     )
     @ddt.unpack
     def test_linked_in_url(self, cert_mode, expected_cert_name):
-        config = LinkedInAddToProfileConfiguration(
-            company_identifier='0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9',
-            enabled=True
-        )
-
-        expected_url = (
-            'http://www.linkedin.com/profile/add'
-            '?_ed=0_mC_o2MizqdtZEmkVXjH4eYwMj4DnkCWrZP_D9&'
-            'pfCertificationName={platform_name}+{expected_cert_name}&'
-            'pfCertificationUrl=http%3A%2F%2Fs3.edx%2Fcert&'
-            'source=o'
-        ).format(
-            expected_cert_name=expected_cert_name,
-            platform_name=quote(settings.PLATFORM_NAME.encode('utf-8'))
-        )
-
-        actual_url = config.add_to_profile_url(
-            self.COURSE_KEY,
-            self.COURSE_NAME,
-            cert_mode,
-            self.CERT_URL
-        )
-
-        self.assertEqual(actual_url, expected_url)
+        config = LinkedInAddToProfileConfigurationFactory()
+
+        # We can switch to this once edx-platform reaches Python 3.8
```

Comment:
Weird, but good callout. Not sure how people will notice, unless they're reading tests for fun though. Wish we knew of a tag or something for these issues.

Reply:
I know :( I just didn't know what else to do. If I didn't put the comment, we would never switch, and we technically don't need to, I just think it is a bit cleaner the other way.